lib/rubocop/cop/style/to_json.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Avoids having both `JSON.generate` and `Object#to_json` in your codebase.
+      # They are equivalent, so for simplicity it makes sense to use only one.
+      # `to_json` is also what ActiveModel::Serializer uses, to it's easier to standardize on it.
+      #
+      # @example
+      #   # bad
+      #   JSON.generate(something)
+      #
+      #   # good
+      #   something.to_json
+      class ToJson < Base
+        extend AutoCorrector
+
+        MSG = 'Use `.to_json` instead of `JSON.generate`.'
+
+        # @!method json_generate?(node)
+        def_node_matcher :json_generate?, '(send (const nil? :JSON) :generate ...)'
+
+        def on_send(node)
+          return unless json_generate?(node)
+
+          add_offense(node) do |corrector|
+            _receiver, _method_name, *args = *node
+            data, opts = handle_args(args)
+            replacement = "#{data}.to_json"
+            replacement += "(#{opts})" if opts
+            corrector.replace(node, replacement)
+          end
+        end
+
+        private
+
+        def handle_args(args)
```

Comment:
If you pass the `node` in instead, there are some useful helpers that could be leveraged (e.g. `first_argument`).